packages/apps/human-app/frontend/src/api/fetch-refresh-token.ts

```diff
@@ -0,0 +1,34 @@
+import { apiPaths } from '@/api/api-paths';
+import { browserAuthProvider } from '@/shared/helpers/browser-auth-provider';
+import { signInSuccessResponseSchema } from '@/api/services/worker/sign-in/schema';
+
+export const fetchTokenRefresh = async (baseUrl: string) => {
+  const response = await fetch(
+    `${baseUrl}${apiPaths.worker.obtainAccessToken.path}`,
+    {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        // eslint-disable-next-line camelcase -- camel case defined by api
+        refresh_token: browserAuthProvider.getRefreshToken(),
+      }),
+    }
+  );
+
+  if (!response.ok) {
+    throw new Error('Failed to refresh token');
+  }
+
+  const data: unknown = await response.json();
+
+  const refetchAccessTokenSuccess = signInSuccessResponseSchema.parse(data);
+
+  browserAuthProvider.signIn(
```

Comment:
Function is called `fetchTokenRefresh`, so it's assumed to have only one purpose: fetch new token, parse response and return it.

The logic with `browserAuthProvider.signIn` (and also `browserAuthProvider.signOut` for errors, that is missed while refactoring) should be in `refreshToken` that calls `fetchTokenRefresh`